eknrenderer/eknrenderer/eknr-renderer.c

```diff
@@ -0,0 +1,1377 @@
+                                                           variant,
+                                                           error);
+}
+
+/**
+ * eknr_renderer_render_content:
+ * @renderer: An #EknrRenderer
+ * @body_html: The underlying HTML body
+ * @server_templated: Whether templating already occurred at build time
+ * @source: Where this content came from
+ * @source_name: Name of the source
+ * @original_uri: URI this content came from
+ * @license: Content license
+ * @title: Content title
+ * @show_title: %TRUE if the article title should be rendered out too.
+ * @use_scroll_manager: %TRUE if the scroll manager should be used, %FALSE otherwise
+ * @error: A #GError
+ *
+ * Render the content and return the rendered content.;
+ *
+ * Returns: (transfer full): A string of rendered HTML or %NULL on error.
+ */
+char *
+eknr_renderer_render_content (EknrRenderer *renderer,
+                              const char   *body_html,
+                              gboolean      server_templated,
+                              const char   *source,
+                              const char   *source_name,
+                              const char   *original_uri,
+                              const char   *license,
+                              const char   *title,
+                              gboolean      show_title,
+                              gboolean      use_scroll_manager,
+                              GError       **error)
+{
+  if (server_templated)
+    return g_strdup (body_html);
+
+  if (g_strcmp0 (source, "wikipedia") == 0 ||
+      g_strcmp0 (source, "wikihow") == 0 ||
+      g_strcmp0 (source, "wikisource") == 0 ||
+      g_strcmp0 (source, "wikibooks") == 0)
+    return _renderer_render_legacy_content(renderer,
+                                           body_html,
+                                           source,
+                                           source_name,
+                                           original_uri,
+                                           license,
+                                           title,
+                                           show_title,
+                                           use_scroll_manager,
+                                           error);
+  else if (g_strcmp0 (source, "prensa-libre") == 0)
+    return _renderer_render_prensa_libre_content(renderer, body_html, error);
+
+  g_set_error (error,
+               EKNR_ERROR,
+               EKNR_ERROR_UNKNOWN_LEGACY_SOURCE,
+               "HTML is not server templated, but no renderer exists for %s",
+               source);
+
+  return NULL;
+}
+
+static GVariant *
+get_wrapper_css_files (const char * const *custom_css_files)
+{
+  GPtrArray *wrapper_css_files = g_ptr_array_new_with_free_func (g_free);
+  g_auto(GStrv) strv = NULL;
+
+  g_ptr_array_add (wrapper_css_files, g_strdup ("clipboard.css"));
+  g_ptr_array_add (wrapper_css_files, g_strdup ("share-actions.css"));
```

Comment:
You don't really need to copy all these strings, do you? `g_variant_new_strv()` will make a copy.

Reply:
Handy, okay, I can avoid the copy then.